src/Compiler/Utilities/range.fs

```diff
@@ -193,28 +193,29 @@ type FileIndexTable() =
     //
     // TO move forward we should eventually introduce a new type NormalizedFileName that tracks this invariant.
     member t.FileToIndex normalize filePath =
-        match fileToIndexTable.TryGetValue filePath with
-        | true, idx -> idx
-        | _ ->
-
-            // Try again looking for a normalized entry.
-            let normalizedFilePath =
-                if normalize then
-                    FileSystem.NormalizePathShim filePath
-                else
-                    filePath
+        lock fileToIndexTable
```

Comment:
How often is this hit for e.g. pars.fs generated file?

Reply:
3718 times.

If this is a concern, we can add an unlocked read-check before the locked operations, so that the lock is entered only once per file.

Reply:
I think that would be good, because we do parallelize parsing.
(or alternatively, somehow demonstrate/measure the added cost)

Reply:
I added the additional unlocked read. Just in case in the future some feature is added that makes the call more contentious.